src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,136 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (first == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (getNode(index) == first) {
+            newNode.next = getNode(index);
+            first = newNode;
+        } else {
+            newNode.prev = getNode(index).prev;
+            newNode.next = getNode(index).next;
+            getNode(index).prev.next = newNode;
+        }
+        getNode(index).prev = newNode;
```

Comment:
dont call the method `getNode` 6 times, just save the reference

Reply:
done